nexus/db-model/src/system_update.rs

```diff
@@ -0,0 +1,168 @@
+    fn from(component_type: UpdateableComponentType) -> Self {
+        match component_type {
+            UpdateableComponentType::BootloaderForRot => Self::BootloaderForRot,
+            UpdateableComponentType::BootloaderForSp => Self::BootloaderForSp,
+            UpdateableComponentType::BootloaderForHostProc => {
+                Self::BootloaderForHostProc
+            }
+            UpdateableComponentType::HubrisForPscRot => Self::HubrisForPscRot,
+            UpdateableComponentType::HubrisForPscSp => Self::HubrisForPscSp,
+            UpdateableComponentType::HubrisForSidecarRot => {
+                Self::HubrisForSidecarRot
+            }
+            UpdateableComponentType::HubrisForSidecarSp => {
+                Self::HubrisForSidecarSp
+            }
+            UpdateableComponentType::HubrisForGimletRot => {
+                Self::HubrisForGimletRot
+            }
+            UpdateableComponentType::HubrisForGimletSp => {
+                Self::HubrisForGimletSp
+            }
+            UpdateableComponentType::HeliosHostPhase1 => Self::HeliosHostPhase1,
+            UpdateableComponentType::HeliosHostPhase2 => Self::HeliosHostPhase2,
+            UpdateableComponentType::HostOmicron => Self::HostOmicron,
+        }
+    }
+}
+
+#[derive(
+    Queryable,
+    Insertable,
+    Selectable,
+    Clone,
+    Debug,
+    Asset,
+    Serialize,
+    Deserialize,
+)]
+#[diesel(table_name = component_update)]
+pub struct ComponentUpdate {
+    #[diesel(embed)]
+    pub identity: ComponentUpdateIdentity,
+    pub version: SemverVersion,
+    pub component_type: UpdateableComponentType,
+    pub parent_id: Option<Uuid>,
+}
+
+#[derive(
+    Queryable, Insertable, Selectable, Clone, Debug, Serialize, Deserialize,
+)]
+#[diesel(table_name = system_update_component_update)]
+pub struct SystemUpdateComponentUpdate {
+    pub component_update_id: Uuid,
+    pub system_update_id: Uuid,
+}
+
+impl From<ComponentUpdate> for views::ComponentUpdate {
+    fn from(component_update: ComponentUpdate) -> Self {
+        Self {
+            identity: component_update.identity(),
+            version: component_update.version.into(),
+            component_type: component_update.component_type.into(),
+            parent_id: component_update.parent_id,
+        }
+    }
+}
+
+#[derive(
+    Queryable,
+    Insertable,
+    Selectable,
+    Clone,
+    Debug,
+    Asset,
+    Serialize,
+    Deserialize,
+)]
+#[diesel(table_name = updateable_component)]
+pub struct UpdateableComponent {
+    #[diesel(embed)]
+    pub identity: UpdateableComponentIdentity,
+    pub device_id: String,
+    pub component_type: UpdateableComponentType,
+    pub version: SemverVersion,
+    // pub status: VersionStatus,
+    /// ID of the parent component, e.g., the sled a disk belongs to. Value will
+    /// be `None` for top-level components whose "parent" is the rack.
+    pub parent_id: Option<Uuid>,
+}
```

Comment:
As mentioned in the "open questions" section of the PR description, I've determined this model is insufficient because it does not allow me to represent nodes like `PSC` or `Host` below (example tree taken from [RFD 334](https://rfd.shared.oxide.computer/rfd/0334#_requirements)), which are parents of updateable components but which are not themselves directly associated with any update artifacts. Displaying this tree in full seems pretty important for the UI, so I'm going to rethink this model a bit. It's probably not a drastic change.

```
Rack─┬─PSC─┬─RoT──Hubris Archive ABCD
     │     └─SP───Hubris Archive DCBA
     ├─Sidecar─┬─RoT──Hubris Archive ABCD
     │         └─SP───Hubris Archive FACE
     ├─Sidecar─┬─RoT──Hubris Archive ABCD
     │         └─SP───Hubris Archive FACE
     ├─Scrimlet─┬─RoT──Hubris Archive ABCD
     │          ├─SP───Hubris Archive BEEF
     │          └─Host─┬─Host Phase 1 Image 1234
     │                 ├─Host Phase 2 Image 1234
     │                 └─Omicron Image 5678
     ├─Scrimlet …
     ├─Gimlet …
     └─Gimlet─┬─RoT──Hubris Archive ABCD
              ├─SP───Hubris Archive BEEF
              └─Host─┬─Host Phase 1 Image 1234
                     ├─Host Phase 2 Image 1234
                     └─Omicron Image 5678
```